fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -645,4 +736,20 @@ private void checkLinkForLdpResourceCreation(final String link) {
         }
     }
 
+    /**
+     * Parse the RFC-3230 Digest response header value.  Look for a
+     * sha1 checksum and return it as a urn, if missing or malformed
+     * an empty string is returned.
+     * @param digest The Digest header value
+     * @return the sha1 checksum value
+     */
+    private String parseDigestHeader(final String digest) {
+        final Map<String,String> digestPairs = RFC3230_SPLITTER.split(nullToEmpty(digest));
+        final String hash = digestPairs.entrySet().stream()
+                    .filter(s -> s.getKey().toLowerCase().equals("sha1"))
+                    .map(Map.Entry::getValue)
+                    .findFirst()
+                    .orElse("");
+        return StringUtils.isBlank(hash) ? "" : "urn:sha1:" + hash;
```

Comment:
Thanks for the pointers, I don't routinely write Java so haven't had much of an opportunity to play with features from 8.